src/zerg/CombatZerg.ts

```diff
@@ -256,18 +263,18 @@ export class CombatZerg extends Zerg {
 		// Fight within the room
 		const target = CombatTargeting.findTarget(this);
 		const preferRanged = this.getActiveBodyparts(RANGED_ATTACK) > this.getActiveBodyparts(ATTACK);
-		const targetRange = preferRanged ? 3 : 1;
+		const targetRange = preferredRange || preferRanged ? 3 : 1;
```

Comment:
having `preferRanged` and `preferredRange` as two different variables is pretty sus 👀